.github/workflows/build.yaml

```diff
@@ -195,7 +195,15 @@ jobs:
                        --target linux-x64-rpc-console \
                        build \
                     "
-
+            - name: Clang-tidy validation
+              timeout-minutes: 20
+              run: |
+                  ./scripts/run_in_build_env.sh \
+                    "./scripts/run-clang-tidy-on-compile-commands.py \
+                       --no-log-timestamps \
+                       --compile-database out/linux-x64-chip-tool-ipv6only/compile_commands.json \
```

Comment:
Why only chip-tool?

Reply:
I had to start somewhere, we can run it on other places as well (all clusters app comes to mind).

Did not try to be exhaustive as that seems not easy as we have a lot of targets. I will add more steps, however I believe in this case supporting multiple compilation databases is probably also desirable. Will try to do that.

Reply:
Updated a bit, requires #16325 since we run clang-tidy over clang variants.

Reply:
I guess I don't understand why the current approach of "run tidy over anything we actually compile" is being changed to one that only looks at a subset of those things, or multiple (possibly overlapping) subsets.

Reply:
Updated to run over the same files as before except restriction of not doing third_party because it is pointless.